src/core/halide_utils.cc

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "tc/core/halide_utils.h"
+
+#include <map>
+#include <vector>
+
+#include "tc/core/flags.h"
+#include "tc/core/tc2halide.h"
+#include "tc/core/utils/dlpack.h"
+
+namespace tc {
+
+using namespace dlutils;
+using namespace Halide;
+using namespace Halide::Internal;
+
+DLDataType fromHalideType(const Halide::Type& type) {
+  // Hmm, these are suspiciously similar...
+  DLDataType dtype;
+  dtype.bits = type.bits();
+  dtype.code = type.code();
+  dtype.lanes = type.lanes();
+  return dtype;
+}
+
+std::map<std::string, int> computeParamValueMap(
+    const tc2halide::HalideComponents& halide,
+    const std::vector<const DLTensor*>& inputsDLT) {
+  std::map<std::string, int> pvm;
+  CHECK_EQ(halide.inputs.size(), inputsDLT.size())
+      << "Mismatched HalideIR and DLTensor number of inputs";
+
+  // fill the ParameterValMap
+  for (size_t i = 0; i < inputsDLT.size(); i++) {
+    const ImageParam& in = halide.inputs[i];
+    const DLTensor* tensor = inputsDLT[i];
+    // for error messages
+    auto param_type = halide.getDef().params()[i].tensorType();
+    for (int d = 0; d < in.dimensions(); d++) {
+      Expr extent = in.parameter().extent_constraint(d);
+      int64_t current_size = tensor->shape[d];
```

Comment:
camelcase please :)

Reply:
yes, it's inherited code but I'll fix it, thanks!